browser/src/layer/tile/CommentListSection.ts

```diff
@@ -375,39 +374,44 @@ export class CommentSection extends CanvasSectionObject {
 		}
 	}
 
-	public newAnnotationVex (comment: any, addCommentFn: any, isMod: any): void {
+	public newAnnotationMobile (comment: any, addCommentFn: any, isMod: any): void {
 		var commentData = comment.sectionProperties.data;
 
-		var dialog = vex.dialog.open({
-			contentClassName: 'vex-has-inputs',
-			message: '',
-			input: [
-				'<textarea name="comment" id="new-mobile-comment-input-area" class="cool-annotation-textarea" required>' + (commentData.text && isMod ? commentData.text: '') + '</textarea>'
-			].join(''),
-			buttons: [
-				$.extend({}, vex.dialog.buttons.NO, { text: _('Cancel') }),
-				$.extend({}, vex.dialog.buttons.YES, { text: _('Save') })
-			],
-			callback: function (data: any) {
-				if (data) {
-					var annotation = comment;
-
-					annotation.sectionProperties.data.text = data.comment;
-					comment.text = data.comment;
-
-					addCommentFn.call(annotation, annotation, comment);
-					if (!isMod)
-						this.containerObject.removeSection(annotation);
-				}
-				else {
-					this.cancel(comment);
-				}
-			}.bind(this),
+		this.map.uiManager.showInputModal('new-annotation-dialog', '', '', '', _('Save'), function(data: string) {
+			if (data) {
+				var annotation = comment;
+
+				annotation.sectionProperties.data.text = data;
+				comment.text = data;
+
+				addCommentFn.call(annotation, annotation, comment);
+				if (!isMod)
+					this.containerObject.removeSection(annotation);
+			}
+			else {
+				this.cancel(comment);
+			}
+		}.bind(this), false);
+
+		document.getElementById('response-cancel').addEventListener('click', function() { this.cancel(comment); }.bind(this));
+
+		document.getElementById('input-modal-input').outerHTML = '<textarea name="comment" id="input-modal-input" class="cool-annotation-textarea" required>' + (commentData.text && isMod ? commentData.text: '') + '</textarea>';
+
+		// Allow close on click away only in desktop.
+		if (document.getElementsByClassName('mobile-wizard jsdialog-overlay cancellable').length > 0) {
+			document.getElementsByClassName('mobile-wizard jsdialog-overlay cancellable')[0].addEventListener('click', function() {
+				this.cancel(comment);
+			}.bind(this));
+		}
+		else {
+			document.getElementById('modal-dialog-new-annotation-dialog-overlay').addEventListener('click', function() {
```

Comment:
It looks a bit like a hack when we create dialog and them manually override callbacks and replacing nodes.
What about using `_modalDialogJSON` function already with textarea within it. Also it is possible to setup callbacks when calling `showModal` function. (Something like internals of `showInfoModal`, adjusted to needs). For me it will look much more clean and would take similar number of lines of code.

Reply:
I implemented that